dataproc/dataproc.go

```diff
@@ -0,0 +1,304 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Command dataproc uses the Cloud Dataproc API to create a cluster,
+// submit a sample job, and delete the cluster when finished.
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+
+	dataproc "google.golang.org/api/dataproc/v1"
+	storage "cloud.google.com/go/storage"
+)
+
+const (
+	// Filename of example job (Python) file uploaded to Cloud Storage
+	exampleJobName = "test-file"
+
+	// Allows for full access to Google Cloud Platform products
+	scope = "https://www.googleapis.com/auth/cloud-platform"
+
+	// Compute URI base
+	computeUriBase = "https://www.googleapis.com/compute/v1/projects/%s/zones/%s"
+)
+
+var (
+	bucketName  = flag.String("bucket", "", "GCS Bucket for storage of job file")
+	clusterName = flag.String("cluster-name", "", "Name of cluster")
+	projectID   = flag.String("project", "", "Your cloud project ID.")
+	pysparkFile = flag.String("pyspark-file", "", "PySpark Job file")
+	region      = flag.String("region", "global", "Your cloud project region.")
+	zoneID      = flag.String("zone", "", "Cloud Platform zone")
+)
+
+// clusterConfig defines the confuration of a cluster including its project, name, and region.
+type clusterConfig struct{
+		bucket,
+    project,
+		region,
+		state,
+		name,
+		uuid,
+		zone string
+}
+
+func main() {
+	// Parse command line arguments
+	flag.Parse()
+
+	// Check for required flags (golang flags does not support?)
+	if *bucketName == "" || *clusterName == "" || *projectID == "" || *pysparkFile == "" || *region == "" || *zoneID == "" {
+		log.Fatal("Incorrect arguments specified see 'go-dataproc -help' for help")
+	}
+
+	// Create a new Cloud Dataproc service
+	ctx := context.Background()
+	client, err := google.DefaultClient(ctx, scope)
+	if err != nil {
+		log.Fatal(err)
+	}
+	service, err := dataproc.New(client)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a new Google Clout Storage service
+	storageService, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a new clusterConfig to hold details about this cluster
+	cluster := clusterConfig{project: *projectID, region: *region, name: *clusterName, zone: *zoneID}
+
+	// Create a cluster
+	if _, err := createCluster(service, &cluster); err != nil {
+		log.Fatal(err)
+	}
+	log.Println("Cluster created")
+
+	// Wait on the cluster to become active
+	log.Println("Waiting for cluster to be ready")
```

Comment:
Again, the comment says the same thing as the log statement. Remove the comments.